### :x: 1 Tests Failed:
| Tests completed | Failed | Passed | Skipped |
|---|---|---|---|
| 27 | 1 | 26 | 8 |
<details><summary>View the top 1 failed test(s) by shortest run time</summary>

> <pre><code class="language-python">tests.test_smartsheet_dataframe.TestGetReportAsDf::test_get_report_as_df_with_both_token_and_report_obj</code></pre>
> <details><summary>Stack Traces | 0.002s run time</summary>
> 
> > <pre><code class="language-python">self = &lt;test_smartsheet_dataframe.TestGetReportAsDf object at 0x7f6dcb6fd350&gt;
> > mock_get_from_request = &lt;MagicMock name=&#x27;_get_from_request&#x27; id=&#x27;140109534981968&#x27;&gt;
> > mock_warn = &lt;MagicMock name=&#x27;warn&#x27; id=&#x27;140109534610000&#x27;&gt;
> > 
> >     @patch(&#x27;warnings.warn&#x27;)
> >     @patch(&#x27;smartsheet_dataframe.smartsheet_dataframe._get_from_request&#x27;)
> >     def test_get_report_as_df_with_both_token_and_report_obj(self, mock_get_from_request, mock_warn):
> >         mock_response = {
> >             &quot;columns&quot;: [{&quot;title&quot;: &quot;Column1&quot;}, {&quot;title&quot;: &quot;Column2&quot;}],
> >             &quot;rows&quot;: [{&quot;id&quot;: 1, &quot;cells&quot;: [{&quot;value&quot;: &quot;Value1&quot;}, {&quot;value&quot;: &quot;Value2&quot;}]}]
> >         }
> >         mock_get_from_request.return_value = mock_response
> >     
> >         mock_report_obj = Mock()
> >         mock_report_obj.to_dict.return_value = mock_response
> >     
> >         df = get_report_as_df(token=&quot;fake_token&quot;, report_id=12345, report_obj=mock_report_obj)
> >     
> > &gt;       mock_warn.assert_called_with(&quot;A &#x27;report_id&#x27; has been provided along with a &#x27;report_obj&#x27; \n&quot; +
> >                                      &quot;The &#x27;sheet_id&#x27; parameter will be ignored&quot;)
> > 
> > tests/test_smartsheet_dataframe.py:115: 
> > _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> > 
> > self = &lt;MagicMock name=&#x27;warn&#x27; id=&#x27;140109534610000&#x27;&gt;
> > args = (&quot;A &#x27;report_id&#x27; has been provided along with a &#x27;report_obj&#x27; \nThe &#x27;sheet_id&#x27; parameter will be ignored&quot;,)
> > kwargs = {}
> > expected = call(&quot;A &#x27;report_id&#x27; has been provided along with a &#x27;report_obj&#x27; \nThe &#x27;sheet_id&#x27; parameter will be ignored&quot;)
> > actual = call(&quot;A &#x27;report_id&#x27; has been provided along with a &#x27;report_obj&#x27; \nThe &#x27;report_id&#x27; parameter will be ignored&quot;)
> > _error_message = &lt;function NonCallableMock.assert_called_with.&lt;locals&gt;._error_message at 0x7f6ddb9bd940&gt;
> > cause = None
> > 
> >     def assert_called_with(self, /, *args, **kwargs):
> >         &quot;&quot;&quot;assert that the last call was made with the specified arguments.
> >     
> >         Raises an AssertionError if the args and keyword args passed in are
> >         different to the last call to the mock.&quot;&quot;&quot;
> >         if self.call_args is None:
> >             expected = self._format_mock_call_signature(args, kwargs)
> >             actual = &#x27;not called.&#x27;
> >             error_message = (&#x27;expected call not found.\nExpected: %s\n  Actual: %s&#x27;
> >                     % (expected, actual))
> >             raise AssertionError(error_message)
> >     
> >         def _error_message():
> >             msg = self._format_mock_failure_message(args, kwargs)
> >             return msg
> >         expected = self._call_matcher(_Call((args, kwargs), two=True))
> >         actual = self._call_matcher(self.call_args)
> >         if actual != expected:
> >             cause = expected if isinstance(expected, Exception) else None
> > &gt;           raise AssertionError(_error_message()) from cause
> > E           AssertionError: expected call not found.
> > E           Expected: warn(&quot;A &#x27;report_id&#x27; has been provided along with a &#x27;report_obj&#x27; \nThe &#x27;sheet_id&#x27; parameter will be ignored&quot;)
> > E             Actual: warn(&quot;A &#x27;report_id&#x27; has been provided along with a &#x27;report_obj&#x27; \nThe &#x27;report_id&#x27; parameter will be ignored&quot;)
> > 
> > ../../../..../uv/python/cpython-3.11.14-linux-x86_64-gnu/lib/python3.11/unittest/mock.py:939: AssertionError</code></pre>
> 
> </details>

</details>

To view more test analytics, go to the [Test Analytics Dashboard](https://app.codecov.io/gh/RCoff/Smartsheet-DataFrame/tests/dev)
<sub>📋 Got 3 mins? [Take this short survey](https://forms.gle/22i53Qa1CySZjA6c7) to help us improve Test Analytics.</sub>

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -2,6 +2,12 @@ name: Run Tests
 
 on: [ push, pull_request ]
 
+env:
+  smartsheet_access_token: ${{ vars.smartsheet_access_token }}
+  testing_report_id: ${{ vars.testing_report_id }}
+  testing_sheet_id: ${{ vars.testing_sheet_id }}
+
+
 jobs:
   test:
     runs-on: ubuntu-latest

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -101,6 +101,7 @@ dev = [
     "pyright>=0.0.13.post0",
     "pytest>=7.0.1",
     "pytest-cov>=4.0.0",
+    "python-dotenv>=0.20.0",
     "ruff>=0.0.17",
     "smartsheet-python-sdk==3.0.5",
 ]

